SecurityPkg/DeviceSecurity/SpdmSecurityLib/SpdmConnectionInit.c

```diff
@@ -403,12 +503,19 @@ CreateSpdmDeviceContext (
 
   SpdmReturn = SpdmSetData (SpdmContext, SpdmDataBaseHashAlgo, &Parameter, &Data32, sizeof (Data32));
   if (LIBSPDM_STATUS_IS_ERROR (SpdmReturn)) {
+    ASSERT (FALSE);
     goto Error;
   }
 
   SpdmReturn = SpdmInitConnection (SpdmContext, FALSE);
   if (LIBSPDM_STATUS_IS_ERROR (SpdmReturn)) {
     DEBUG ((DEBUG_ERROR, "SpdmInitConnection - %p\n", SpdmReturn));
+
+    AuthState = TCG_DEVICE_SECURITY_EVENT_DATA_DEVICE_AUTH_STATE_NO_SPDM;
+    Status = ExtendNoSpdm (SpdmDeviceContext, AuthState);

```

Comment:
The measure for `AUTH_NO_SPDM + No Data` is same as `AUTH_FAIL_NO_SIG + No Data`, I think maybe here we can reuse the `ExtendCertificate(...)` function, no need to create this new function `ExtendNoSpdm(...)`.

Reply:
Thanks. I have changed the code.